test/e2e/servers/graphql-main-server/with-auth-requests.spec.ts

```diff
@@ -336,6 +353,36 @@ describe("graphql", () => {
     })
   })
 
+  describe("lnUsdInvoiceCreate", () => {
+    const mutation = LN_USD_INVOICE_CREATE
+
+    it("returns a valid lightning invoice", async () => {
+      const input = {
+        walletId: usdWalletId,
+        amount: 1000,
+        memo: "This is a lightning invoice",
+      }
+      const result = await apolloClient.mutate({ mutation, variables: { input } })
+      const { invoice, errors } = result.data.lnUsdInvoiceCreate
+      expect(errors).toHaveLength(0)
+      expect(invoice).toHaveProperty("paymentRequest")
+      expect(invoice.paymentRequest.startsWith("lnbcrt4")).toBeTruthy()
```

Comment:
shouldn't we check that the amount in sats is the correct one?

Reply:
That would couple this test to the business logic of the dealer. The purpose of e2e tests is to assert that all cross service boundary communication driven by the public interface are working.

The dealer business logic is being tested in https://github.com/GaloyMoney/stablesats-rs/blob/698cb07c56625ecb3e2924f2ec27feea5f105ff6/price-server/tests/price_app.rs#L20. If it changes we don’t want to break this test.

Reply:
In general we are using these e2e tests for too much. Imo e we could probably delete quite a few completely or move the assertions into integration and unit tests. 

That would reduce our testing runtime while keeping the same correctness guarantees

Reply:
integration tests are currently taking longer than e2e tests

Reply:
> The dealer business logic is being tested in https://github.com/GaloyMoney/stablesats-rs/blob/698cb07c56625ecb3e2924f2ec27feea5f105ff6/price-server/tests/price_app.rs#L20. If it changes we don’t want to break this test.

that makes sense

Reply:
btw the `4` in `lnbcrt4` still implies some sats amount been returned (just the first digit)

Reply:
> integration tests are currently taking longer than e2e tests
=> unit tests FTW!
> btw the 4 in lnbcrt4 still implies some sats amount been returned (just the first digit)
Okay I used the same number of chars as for the other tests. I doubt the first digit would change - but if it does and the test breaks we can remove it.

Reply:
Based on the fix from #1558 this test should've failed here (see [BOLT-11 spec](https://github.com/lightning/bolts/blob/master/11-payment-encoding.md#human-readable-part)). Why did it pass?